### ISSUE TITLE
Ensure that we don't miss processing `rust-dlc` transactions

### ISSUE THREAD

**Issue by luckysori** [get10101/10101#790]
We had to open https://github.com/get10101/10101/pull/789 because some tests were failing.

At the same time, it seems like in theory we could miss processing transactions in the `SubChannelManager::periodic_check`. See https://github.com/p2pderivatives/rust-dlc/commit/4e85d15 (the patch that we just reverted) for more context.

**Comment by klochowicz** [comment]
shall we discuss the path forward with Tibo?

**Comment by luckysori** [comment]
> shall we discuss the path forward with Tibo?

We've had a couple of offline chats about it with him already.

Originally we set out to rewrite the way we feed blocks into `rust-dlc`'s `ChainMonitor`, but the change was a bit more involved than what we desired so we left it.

Since then we've also considered using `lightning-transaction-sync` in order to not have to maintain a separate chain monitor for `rust-dlc`, but it is assumed that it's not a straightforward change (this assumption needs to be checked).
